### PR TITLE
Completely revamp vector-vector binop matching visualizer

### PR DESCRIPTION
**Pull request by juliusv**
This completely reimplements the vector-to-vector binop visualizer in a (hopefully) prettier, cleaner, and more powerful way. Main highlights:

* Everything is (hopefully) prettier now, with match groups being displayed in colored boxes (instead of one big table), and series within match groups having a color-coded swatch that is tracked into the result.
* The logic for calculating the match groups is now separated out (and completely reimplemented) in the `promql` "package" and could be reused by other things now.
* Match groups are now also shown for error cases, and the error cases are called out and explained more clearly (per match group).
* There are now safety limits both for the number of displayed groups and the number of series per group (both currently at 100 max). Both can be turned off with a button when needed.
* The sample values for the series can be switched on/off and are off by default, since they are not usually needed to figure out matching behavior.

One-to-one matching example:

image

One-to-one matching with some missing matches:

image

Many-to-one matching:

image

No `group_left()` modifier for many-to-one matching error:

image

Using `group_right()` when `group_left()` would be needed:

image

### CODE REVIEW

File: app/src/promql/binOp.ts [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+  }
+): BinOpResult => {
+  // For the simplification of further calculations, we assume that the "one" side of a one-to-many match
+  // is always the right-hand side of the binop and swap otherwise to ensure this. We swap back in the end.
+  [lhs, rhs] = matching.card === vectorMatchCardinality.oneToMany ? [rhs, lhs] : [lhs, rhs];
+
+  const groups: BinOpMatchGroups = {};
+  const sigf = signatureFunc(matching.on, matching.labels);
+
+  const outOfLimitGroups = new Set<string>();
```
Comment (Nexucis) [inline app/src/promql/binOp.ts:254]: you could just have an integer variable here to increase each time you would drop something.
It doesn't look like you are using this set except for its size

Comment (Nexucis) [reply app/src/promql/binOp.ts:254]: ah yes you are right. Perhaps you could add a little comment to avoid the mistake later

Comment (juliusv) [reply app/src/promql/binOp.ts:257]: Done!

**Review by Nexucis — APPROVED** [review]
perfect 🥳

**Pull request merged**